angular/src/app/player-detail.component.ts

```diff
@@ -0,0 +1,19 @@
+import { Component, Input } from '@angular/core';
+import { Player } from './player';
+
+@Component({
+  selector: 'player-detail',
+  template: `
+    <div *ngIf="player">
```

Comment:
It seems strange to me to have this component be responsible for not showing itself if there is no player. It would make more sense IMO to make that the job of the parent component, and this just always shows its player. The `ngIf` would go in the parent template instead.